base/pkg/read.jl

```diff
@@ -3,18 +3,29 @@
 module Read
 
 import ...LibGit2, ..Cache, ..Reqs, ...Pkg.PkgError, ..Dir
-using ..Types
+using ..Types, Base.Filesystem
 
-readstrip(path...) = strip(readstring(joinpath(path...)))
+isgitrepo(pkg) =
+    ispath_casesensitive(joinpath(pkg, ".git"))
```

Comment:
This doesn't actually accomplish anything on MacOS, because `ispath_casesensitive` only checks the `basename` of the path.   So, e.g. `ispath_casesensitive("foo/.git")` will succeed even if the actual directory name is `"Foo"`.